Use `.Release.Namespace` by default to handle namespaces

jeantil opened the pull request:
It remains possible to override the current release namespace by setting
the `namespace` value though this may lead to having the helm metadata
and the pulsar components in different namespaces

Fixes #66

### Motivation

Trying to deploy the chart in a namespace using the usual helm pattern fails for example
```
kubectl create ns pulsartest
helm upgrade --install pulsar -n pulsartest apache/pulsar
Error: namespaces "pulsar" not found
```
fixing that while keeping the helm metadata and the deployed objects in the same namespace requires declaring the namespace twice 
```
kubectl create ns pulsartest
helm upgrade --install pulsar -n pulsartest apache/pulsar --set namespace=pulsartest
Error: namespaces "pulsar" not found
```
This is needlessly confusing for newcomers who follow the helm documentation and is contrary to helm best practices.

### Modifications

I changed the chart to use the context namespace `.Release.Namespace` by default while preserving the ability to override that by explicitly providing a namespace on the commande line, with the this modification both  examples behave as expected
 
### Verifying this change

- [x] Make sure that the change passes the CI checks.

jeantil commented:
@wolfstudy I assume that the ci fails because I don't belong to a specific organization, not sure how I can fix it
```
Digest: sha256:b0d5e8220d7f0594952dc1bd452ce3c9c0a724484320645077eccf610648b88b
Status: Downloaded newer image for gcr.io/google-containers/kube-scheduler:v1.14.10
gcr.io/google-containers/kube-scheduler:v1.14.10
ERROR: no nodes found for cluster "pulsar-ci-1e799b03-a506-4be2-a1e3-117912596fe9"
```

sijie commented on charts/pulsar/templates/bookkeeper-configmap.yaml:
```diff
@@ -22,7 +22,7 @@ apiVersion: v1
 metadata:
   name: "{{ template "pulsar.fullname" . }}-{{ .Values.bookkeeper.component }}"
-  namespace: {{ .Values.namespace }}
+  namespace: {{ .Values.namespace | default .Release.Namespace }}
```
Can you implement this using a template function? 

Like what we did in https://github.com/streamnative/charts/blob/master/charts/pulsar/templates/_helpers.tpl#L21

jeantil replied:
sure, see updated commit

sijie commented:
regarding the CI, can you rebase your pull request to latest master to include the fix in #76 ?

jeantil commented:
I switched to the template function and rebased on master

jeantil commented:
I'm not familiar with charts linting, it appears the error is 
```
 ✖︎ pulsar => (version: "2.6.1-3", path: "charts/pulsar") > Error validating maintainer 'The Apache Pulsar Team': 404 Not Found
```
and I don't think I touched that :D

sijie commented on charts/pulsar/Chart.yaml:
```diff
@@ -21,11 +21,11 @@ apiVersion: v1
 home: https://pulsar.apache.org
 sources:
-- https://github.com/apache/pulsar
+  - https://github.com/apache/pulsar
```
Any ideas on these changes?

jeantil replied:
sorry about that, It's from the prettier plugin in my vscode install, I was pretty sure I had it disabled for this workspace and was careless when adding the modifications. I have reverted the indentation changes

sijie commented on charts/pulsar/values.yaml:
```diff
@@ -372,8 +373,8 @@ bookkeeper:
     ##
     resources:
       # requests:
-        # memory: 4Gi
```
Any ideas on these indention changes?

wolfstudy commented:
@jeantil After https://github.com/apache/pulsar-helm-chart/pull/83 is merged, please merge the latest master code to ensure that ci passes. Thanks.

sijie commented:
@jeantil Can you rebase this pull request to the latest master?

jeantil commented:
@sijie @wolfstudy I rebased on master and incremented the version to 2.6.2-2

The lint task is green but I think it actually failed, the following comes up as a warning in the action
>Unexpected input(s) 'command', valid inputs are ['version']

The rest of the CI chain seemed stuck, after digging in a bit I realized it depended on the namespace being defined in the default values file (not overriden in the `.ci/cluster/` value files  and didn't pass the target namespace to helm.
Because of my patch the deployed containers ended up in the  `default` namespace but the verification commands using kubectl  explicitely queried the `pulsar` namespace.\

I have updated the ci script to explicitely pass the target namespace to helm. let me know if you prefer it another way ...

jeantil commented:
@sijie @wolfstudy Sorry to bother you, it's been a couple weeks and I am worndering if there is more work I need to do on the PR to get it merged (and to fix the CI in the process) ? thanks !

sijie commented:
@jeantil sorry that I missed your comment.

jeantil commented:
Thank you @sijie !
